src/Features/LanguageServer/Protocol/Handler/SemanticTokens/SemanticTokensEditsHandler.cs

```diff
@@ -0,0 +1,377 @@
+            {
+                Start = start,
+                DeleteCount = deleteCount,
+                Data = data
+            };
+
+        private sealed class LongestCommonSemanticTokensSubsequence : LongestCommonSubsequence<SemanticToken[]>
+        {
+            private static readonly LongestCommonSemanticTokensSubsequence s_instance = new LongestCommonSemanticTokensSubsequence();
+
+            protected override bool ItemsEqual(
+                SemanticToken[] oldSemanticTokens, int oldIndex,
+                SemanticToken[] newSemanticTokens, int newIndex)
+                => oldSemanticTokens[oldIndex].Equals(newSemanticTokens[newIndex]);
+
+            public static IEnumerable<SequenceEdit> GetEdits(
+                SemanticToken[] oldSemanticTokens, SemanticToken[] newSemanticTokens)
+                => s_instance.GetEdits(oldSemanticTokens, oldSemanticTokens.Length, newSemanticTokens, newSemanticTokens.Length);
+        }
+
+        /// <summary>
+        /// Stores the values that make up the LSP representation of an individual semantic token.
+        /// </summary>
+        private readonly struct SemanticToken
+        {
+            private readonly int _deltaLine;
+            private readonly int _deltaStartCharacter;
+            private readonly int _length;
+            private readonly int _tokenType;
+            private readonly int _tokenModifiers;
+
+            public SemanticToken(int deltaLine, int deltaStartCharacter, int length, int tokenType, int tokenModifiers)
+            {
+                _deltaLine = deltaLine;
+                _deltaStartCharacter = deltaStartCharacter;
+                _length = length;
+                _tokenType = tokenType;
+                _tokenModifiers = tokenModifiers;
+            }
+
+            public int[] ToArray()
+            {
+                return new int[] { _deltaLine, _deltaStartCharacter, _length, _tokenType, _tokenModifiers };
+            }
+
+            public override bool Equals(object? obj)
+            {
+                return obj is SemanticToken token &&
+                       _deltaLine == token._deltaLine &&
+                       _deltaStartCharacter == token._deltaStartCharacter &&
+                       _length == token._length &&
+                       _tokenType == token._tokenType &&
+                       _tokenModifiers == token._tokenModifiers;
+            }
+
+            public override int GetHashCode()
+            {
+                return Hash.Combine(_deltaLine.GetHashCode(),
+                    Hash.Combine(_deltaStartCharacter.GetHashCode(),
+                    Hash.Combine(_length.GetHashCode(),
+                    Hash.Combine(_tokenType.GetHashCode(),
+                    _tokenModifiers.GetHashCode()))));
+            }
+        }
+
+        [Flags]
+        private enum SemanticTokenEditKind
+        {
+            None = 0,
+            Insert = 1,
+            Delete = 2,
```

Comment:
nit.  you mark this as flags (which is good).  but i still recommend writing this as: 
```suggestion
            Delete = 1 << 1,
```

the bitshift helps make it very clear that these are bit values (And thus intended to be 'or-ed').  That said, as per an above discussion, i think these coudl be non-flags, and just have 3 values (Insert/Delete/Update).

Reply:
Yep, just added an Update value per your suggestion. I kept the None value for a total of 4 values since it's used for some logic in the code.